### PR TITLE
Remove ceph-admin relation, use ceph-client instead

### PR DESCRIPTION
**Pull request by Cynerva**
Fixes https://bugs.launchpad.net/bugs/1840260 and https://bugs.launchpad.net/bugs/1840492 for ceph-csi charm.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ops >= 1.2.0
 kubernetes
 jinja2
-pyyaml
\ No newline at end of file
+pyyaml
+git+https://opendev.org/openstack/charm-ops-interface-ceph-client@d4d0cefc5e92edd28cdbf5d00e190d4498896418#egg=interface_ceph_client
+netifaces
```
Comment (addyess) [inline requirements.txt:6]: What's this for?  I can't find it used in this PR.  Does the above requirement need it but not list it in the `setup.py`

Comment (addyess) [reply requirements.txt:6]: oh, I see it's used in the [`test-requirements.txt`](https://opendev.org/openstack/charm-ops-interface-ceph-client/src/branch/master/test-requirements.txt#L7) of that upstream library.  It's clear i don't quite grok it's necessity but i'll +1

File: src/charm.py [hunk 2]
```diff
@@ -125,14 +127,81 @@ def copy_stored_dict(stored_dict: StoredDict) -> dict:
...
         """Return context that can be used to render ceph resource files in templates/ folder."""
-        raw_mon_hosts = self._stored.ceph_data.get("mon_hosts", "").split()
         return {
-            "fsid": self._stored.ceph_data.get("fsid"),
-            "kubernetes_key": self._stored.ceph_data.get("key"),
-            "mon_hosts": json.dumps(raw_mon_hosts),
+            "fsid": self.get_ceph_fsid(),
+            "kubernetes_key": self.key,
+            "mon_hosts": json.dumps(self.mon_hosts),
+            "user": self.app.name,
```
Comment (addyess) [inline src/charm.py:204]: so, the template value will change from `admin` to `ceph-csi` or whatever this app is named?  Am i reading that right

**Review by addyess — APPROVED** [review]
Questions -- but looks really good.  👍🏻

File: src/charm.py (hunk 2 above)
Comment (Cynerva) [reply src/charm.py:204]: Yeah, that's right. The `ceph-client` relation gives you an auth key for a user named `<application-name>`. The username isn't in the relation data at all, it's just something you have to know when using the ceph-client relation.

File: requirements.txt (hunk 1 above)
Comment (Cynerva) [reply requirements.txt:6]: Oh yeah! Let me show you what happens when this is missing:

```
$ tox -e unit
...
Reading package lists... Done
E: Could not open lock file /var/lib/apt/lists/lock - open (13: Permission denied)
E: Unable to lock directory /var/lib/apt/lists/
W: Problem unlinking the file /var/cache/apt/pkgcache.bin - RemoveCaches (13: Permission denied)
W: Problem unlinking the file /var/cache/apt/srcpkgcache.bin - RemoveCaches (13: Permission denied)
juju-log: Couldn't acquire DPKG lock. Will retry in 10 seconds
^CE..........................
======================================================================
ERROR: test_charm (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_charm
Traceback (most recent call last):
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/charmhelpers/contrib/network/ip.py", line 38, in <module>
    import netifaces
ModuleNotFoundError: No module named 'netifaces'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/charmhelpers/fetch/ubuntu.py", line 922, in _run_with_retries
    result = subprocess.check_call(cmd, env=env, **kwargs)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['apt-get', 'update']' returned non-zero exit status 100.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/tests/unit/test_charm.py", line 27, in <module>
    from interface_ceph_client.ceph_client import CephClientRequires
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/interface_ceph_client/ceph_client.py", line 8, in <module>
    import charmhelpers.contrib.storage.linux.ceph as ch_ceph
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/charmhelpers/contrib/storage/linux/ceph.py", line 69, in <module>
    from charmhelpers.contrib.openstack.utils import config_flags_parser
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/charmhelpers/contrib/openstack/utils.py", line 34, in <module>
    from charmhelpers.contrib.network import ip
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/charmhelpers/contrib/network/ip.py", line 40, in <module>
    apt_update(fatal=True)
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/charmhelpers/fetch/ubuntu.py", line 400, in apt_update
    _run_apt_command(cmd, fatal)
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/charmhelpers/fetch/ubuntu.py", line 945, in _run_apt_command
    _run_with_retries(
  File "/home/gkk/work/tasks/bug-ceph-csi/ceph-csi-operator/.tox/unit/lib/python3.8/site-packages/charmhelpers/fetch/ubuntu.py", line 929, in _run_with_retries
    time.sleep(CMD_RETRY_DELAY)
KeyboardInterrupt
```

Bwuh? An apt command is running, and before the tests even started? Yep, importing `interface_ceph_client` takes us on a charmhelpers import trail that eventually leads to apt calls [here](https://github.com/juju/charm-helpers/blob/a72931f7324526e7f05221847437238517f38fa7/charmhelpers/contrib/network/ip.py#L37-L42).

I don't want that happening in the tests *or* at run time, so I put it in requirements.txt. Hope that makes sense.

Comment (addyess) [reply requirements.txt:6]: FUN!  glad you caught that one

Comment (Cynerva) [reply requirements.txt:9]: I added a comment to hopefully clarify this for future codebase readers, please take a look.

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,9 @@
...
 kubernetes
 jinja2
-pyyaml
\ No newline at end of file
+pyyaml
+git+https://opendev.org/openstack/charm-ops-interface-ceph-client@d4d0cefc5e92edd28cdbf5d00e190d4498896418#egg=interface_ceph_client
+
+# require netifaces to prevent a dynamic apt install during interface_ceph_client -> charmhelpers import
+# https://github.com/juju/charm-helpers/blob/a72931f7324526e7f05221847437238517f38fa7/charmhelpers/contrib/network/ip.py#L37-L42
+netifaces
```
Comment (addyess) [inline requirements.txt:9]: 💯 love this tremendously

**Review by addyess — APPROVED** [review]

**Pull request merged**